s3prl/downstream/a2o-vc-vcc2020/README.md

```diff
@@ -0,0 +1,92 @@
+The first task is intra-lingual VC, and the second task is cross-lingual VC.
+For more details, please refer to the original paper:
+[Yi, Z., Huang, W., Tian, X., Yamagishi, J., Das, R.K., Kinnunen, T., Ling, Z., Toda, T. (2020) Voice Conversion Challenge 2020 –- Intra-lingual semi-parallel and cross-lingual voice conversion –-. Proc. Joint Workshop for the Blizzard Challenge and Voice Conversion Challenge 2020, 80-98, DOI: 10.21437/VCC_BC.2020-14.](https://www.isca-speech.org/archive_v0/VCC_BC_2020/pdfs/VCC2020_paper_13.pdf)
+
+## Implementation
+
+We implement three models. Two of them resemble the the top systems in VCC2018 and VCC2020, as described in the following papers:
+[Liu, L., Ling, Z., Jiang, Y., Zhou, M., Dai, L. (2018) WaveNet Vocoder with Limited Training Data for Voice Conversion. Proc. Interspeech 2018, 1983-1987, DOI: 10.21437/Interspeech.2018-1190.](https://www.isca-speech.org/archive/Interspeech_2018/pdfs/1190.pdf)  
+[Liu, L., Chen, Y., Zhang, J., Jiang, Y., Hu, Y., Ling, Z., Dai, L. (2020) Non-Parallel Voice Conversion with Autoregressive Conversion Model and Duration Adjustment. Proc. Joint Workshop for the Blizzard Challenge and Voice Conversion Challenge 2020, 126-130, DOI: 10.21437/VCC_BC.2020-17.](https://www.isca-speech.org/archive_v0/VCC_BC_2020/pdfs/VCC2020_paper_17.pdf)
+
+We made several modifications.
+1. **Input**: instead of using the bottleneck features (BNFs) of a pretrained ASR model, we use the various upstreams provided in S3PRL.
+2. **Output**: instead of using acoustic features extracted using a high-quality vocoder, STRAIGHT, we use the log-melspectrograms.
+3. **Data**: we benchmark on the [VCC2020](https://github.com/nii-yamagishilab/VCC2020-database) dataset. 
+4. **Training strategy**: instead of pretraining on a multispeaker dataset first, we directly trained on the target speaker training set.
+5. **Vocoder**: instead of using the WaveNet vocoder, we used the [Parallel WaveGAN](https://arxiv.org/abs/1910.11480) (PWG) based on the [open source project](https://github.com/kan-bayashi/ParallelWaveGAN) developed by [kan-bayashi](https://github.com/kan-bayashi).
+
+## Dependencies:
+
+- `parallel-wavegan`
+- `fastdtw`
+- `pyworld`
+- `pysptk`
+- `jiwer`
+- `resemblyzer`
+
+## Usage
+
+### Preparation
+```
+# Download the VCC2020 dataset.
+cd <root-to-s3prl>/s3prl/downstream/a2o-vc-vcc2020
+cd data
+./data_download.sh vcc2020/
+cd ../
+
+# Download the pretrained PWGs.
+./pwg_download.sh ./
+```
+
+### Training
+Here we train an A2O VC model using the `config_ar_taco2.yaml` configuration. Please replace the `<trgspk>`, `<upstream>` in the following command to your desired settings.
+```
+cd <root-to-s3prl>/downstream/a2o-vc-vcc2020
+python ../../run_downstream.py -m train --config config_ar_taco2.yaml -n a2o_vc_vcc2020_<trgspk>_<upstream> -u <upstream> -d a2o-vc-vcc2020 -o "config.trgspk='<trgspk>'"
```

Comment:
I notice that `config.trgspk` does not appear in the config file. I am thinking will it be nice to add placeholder in the example config files? like simply `trgspk: TEF1` and a few lines describing where people can find other speaker codes.

Reply:
I think it will be better to put all downstream-specific configs under the `downstream_expert` field. Gathering configurations together might make `DownstramExpert` more modularized and will make future disentanglement from `run_downstream.py` script easier. Like, someday people might want all these standalone task-modules and train them jointly with another multi-task script.

Reply:
I find that actually due to the default config path, it will be more recommended to run the `run_downstream.py` script under `s3prl/s3prl` instead of `s3prl/s3prl/run_downstream.py`.

Reply:
>  am thinking will it be nice to add placeholder in the example config files? like simply trgspk: TEF1

Sure. I can change this.

> I think it will be better to put all downstream-specific configs under the downstream_expert field.

Sorry, I don't understand what you suggest here...

> it will be more recommended to run the run_downstream.py script under s3prl/s3prl instead of s3prl/s3prl/downstream/vc_folder.

I agree to run `run_downstream.py` in `s3prl/s3prl` if users want to train only one downstream model. 
How about `batch_vc_train.sh`? Originally I put it in `s3prl/s3prl`, but I think that kind of breaks the structure, so I moved `batch_vc_train.sh` to `s3prl/s3prl/downstream/vc_folder`. Then I thought it might be confusing if a single dry run is done in `s3prl/s3prl` but batch run is in `s3prl/s3prl/downstream/vc_folder`... Should we do it this way?

Reply:
> I think it will be better to put all downstream-specific configs under the downstream_expert field.

I think I understand now. Do you mean to put the `trgspk` field in the `downstream_expert` field, like `"config.downstream_expert.trgspk='<trgspk>'"`?

Reply:
> > I think it will be better to put all downstream-specific configs under the downstream_expert field.
> 
> I think I understand now. Do you mean to put the `trgspk` field in the `downstream_expert` field, like `"config.downstream_expert.trgspk='<trgspk>'"`?

Yes! This is what I mean :)

Reply:
> I agree to run `run_downstream.py` in `s3prl/s3prl` if users want to train only one downstream model.
> How about `batch_vc_train.sh`? Originally I put it in `s3prl/s3prl`, but I think that kind of breaks the structure, so I moved `batch_vc_train.sh` to `s3prl/s3prl/downstream/vc_folder`. Then I thought it might be confusing if a single dry run is done in `s3prl/s3prl` but batch run is in `s3prl/s3prl/downstream/vc_folder`... Should we do it this way?

Oh! I can understand your concern... I think actually both ways work (people can either execute at the `s3prl/s3pr/` or inside the vc folder) and it is just about how we document it and *what is the path of the **data folder** in configs*. I think the major policy is that I would like to keep the *benchmarking* usage consistent. That is, the specific setting for benchmarking new representations is able to run from `s3prl/s3prl` as default so that people can run through all the tasks at `s3prl/s3prl`. In your case I think it is fine to put more functions inside the vc folder which is great! I will suggest that we can divide the README into these parts:

1. How to benchmark new upstream with VC task with a simple command. (execute at `s3prl/s3prl`)
2. How to do VC-related experiments provided by this VC folder. (execute at VC folder)

We can even provide different config files with different default path of the **data folder**.
Do you think we can go this way?

Reply:
> I think it is great to put more functions inside the vc folder and I will suggest that we can divide the README into two parts:

> We can even provide different config files with different default path of the data folder. Do you think we can go this way?

Sure! That sounds perfect to me! I think I will create 4 config files:

- `config.yaml`: the default config to benchmark a new upstream, which contains **relative** paths, so needs to be executed in `s3prl/s3prl`.
- `config_simple.yaml`, `config_simple_ar.yaml` and `config_taco2_ar.yaml`: the configs for reproducing experiments in my own VC paper, which contains **absolute** paths, so can be executed in the VC folder.

Reply:
Cool! This also sounds perfect to me.